### ISSUE TITLE
REST api?

### ISSUE THREAD

**Issue by Gsuz** [invoice-x/invoice2data#188]
Has anyone done any work toward building a REST api around this script? It shouldn't be much effort. Wanted to check before starting on my own.

**Comment by m3nu** [comment]
Not that I know of. If you mean a full hosted service, there are a few similar commercial ones. Matching them would be a major effort though. You'd probably need a new way to match fields, as mosts users won't use Regex.

**Comment by m3nu** [comment]
Whats your use-case for this? Using the Python lib with something else?

I can see some use in a web service for making the lib easier to use and involving more people to make templates. But I also worry that without a different system to make templates/teach it what to extract it will still be too hard to use.

**Comment by Gsuz** [comment]
Yes. Basically to handle query's from a web app which has a bunch of invoices to autofill some forms.

**Comment by m3nu** [comment]
I see. So it would be a microservice.

If you have some code, we could host it under the invoice-x organization, and link it in some places, so others can improve and use it. Would just need to have some kind of permissive license and a README, so new contributors can find their way around. I can later set up testing and other admin stuff.

**Comment by gwendall** [comment]
+1 

Any news about that ?

**Comment by GeorgeMihalache** [comment]
Any news?

**Comment by rmilecki** [comment]
I have & use **trivial** `Flask` based web API, I'll try to clean it up and share next month.

**Comment by GeorgeMihalache** [comment]
> I have & use **trivial** `Flask` based web API, I'll try to clean it up and share next month

This is very good news for those unfamiliar with the python environment. Thanks
